R/nest.R

```diff
@@ -329,20 +329,9 @@ unnest.data.frame <- function(
                               .id = "DEPRECATED",
                               .sep = "DEPRECATED",
                               .preserve = "DEPRECATED") {
-
   cols <- tidyselect::eval_select(enquo(cols), data)
-
-  if (nrow(data) == 0) {
-    for (col in names(cols)) {
-      data[[col]] <- as_empty_df(data[[col]], col = col)
-    }
-  } else {
-    for (col in names(cols)) {
-      data[[col]] <- map(data[[col]], as_df, col = col)
-    }
-  }
-
   data <- unchop(data, any_of(cols), keep_empty = keep_empty, ptype = ptype)
+  cols <- cols[map_lgl(unclass(data)[cols], is.data.frame)]
   unpack(data, any_of(cols), names_sep = names_sep, names_repair = names_repair)
```

Comment:
If `unpack()` was smart enough to "skip" any non df-cols (like how unchop "skips" non list-cols) then we wouldn't even need this filtering line! For example, this `unpack()` call _could_ return its input unchanged.

``` r
library(tidyr)

df <- tibble(x = 1L, y = 1)

# "skips" unchopping y
unchop(df, y)
#> # A tibble: 1 x 2
#>       x     y
#>   <int> <dbl>
#> 1     1     1

# requires df cols
unpack(df, y)
#> Error: `y` must be a data frame column
```

That would make `unnest()` truly an `unchop()` + `unpack()` without any intermediate adjustments, which theoretically is kind of nice.

Reply:
This would also be great to easily unpack all data frame columns `unpack(df, everything())` and in general less need to worry when using selection helpers